### PR TITLE
feat: new dropdown multiselect on Rooms page

### PR DESCRIPTION
**Pull request by felipe-rod123**
## Proposed changes (including videos or screenshots)

* Created a new multi-select dropdown component (`apps/meteor/client/views/admin/rooms/DropDown/CustomDropDown.tsx`) to add section titles and other features, as needed. This could be reused to create more variants in the future, or even a new component on Fuselage, maybe.

* This new component can be easily grouped with other dropdowns instances, to create a complex multi-select filter.

* Created a new custom hook to filter rooms, using a Set object to store unique values inside an array, instead of creating a custom function to remove duplicates.

* Removed previous checkbox filters, and added two new multi-select dropdowns to the Admin Rooms page (`apps/meteor/client/views/admin/rooms/RoomsTable.tsx`)

* Added a new "Group" category, to represent private channels.

## Issue(s)

Jira task: [WM-25](https://rocketchat.atlassian.net/browse/WM-25?atlOrigin=eyJpIjoiZjU5ZGJmMDQxMmIzNDgyOWJlM2EzMzY0Y2I4ZDUwMDgiLCJwIjoiaiJ9)

## Steps to test or reproduce

Navigate to: Home → Workspace → Rooms.

## Further comments

WM-25

[WM-25]: https://rocketchat.atlassian.net/browse/WM-25?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by dougfabris — CHANGES_REQUESTED** [review]

File: yarn.lock [hunk 1]
```diff
@@ -1020,30 +1020,30 @@ __metadata:
   languageName: node
   linkType: hard
 
-"@babel/core@npm:^7.1.0, @babel/core@npm:^7.11.6, @babel/core@npm:^7.12.10, @babel/core@npm:^7.12.3, @babel/core@npm:^7.7.5":
```
Comment (hugocostadev) [inline yarn.lock:1023]: Why did you have to update the `yarn.lock` , is that correct?

File: apps/meteor/client/providers/LayoutProvider.tsx [hunk 2]
```diff
@@ -35,8 +35,7 @@ const LayoutProvider: FC = ({ children }) => {
 					},
 					size: {
 						sidebar: '240px',
-						// eslint-disable-next-line no-nested-ternary
-						contextualBar: breakpoints.includes('sm') ? (breakpoints.includes('xl') ? '38%' : '380px') : '100%',
+						contextualBar: breakpoints.includes('sm') ? '380px' : '100%',
```
[the remark this answers is not in the document]
Comment (felipe-rod123) [reply apps/meteor/client/providers/LayoutProvider.tsx:38]: Hmm, I think this was a mistake. I had several conflicts in this PR a few days ago, maybe this got merged by accident... I'll fix that!

**Review by hugocostadev — CHANGES_REQUESTED** [review]
Comparing with the Figma design:
image

I think that the inputs on this page should be smaller and the options should take the entire input, WDYT?

File: .changeset/dropdown.md [hunk 3]
```diff
@@ -0,0 +1,9 @@
+---
+"@rocket.chat/meteor": patch
+---
+
+Created new dropdown filters on the Admin Rooms page:
+
+- added capability for admin users to filter by Room type or Room visibility.
+- removed previous checkbox filters to improve UX.
+- created a new component for the multiselect dropdown.
```
Comment (hugocostadev) [inline .changeset/dropdown.md:9]: This should be user-friendly and release focus.

```suggestion
New filters to the Rooms Table at `Workspace > Rooms`
```

**Review by hugocostadev — CHANGES_REQUESTED** [review]

**Review by hugocostadev — APPROVED** [review]

File: apps/meteor/client/views/admin/rooms/RoomsTable.tsx [hunk 4]
```diff
@@ -205,9 +223,21 @@ const RoomsTable = ({ reload }: { reload: MutableRefObject<() => void> }): React
 		[mediaQuery, onClick, t],
 	);
 
+	function intersect(array1: IRoom[], array2: IRoom[]) {
```
Comment (rique223) [inline apps/meteor/client/views/admin/rooms/RoomsTable.tsx:226]: Please, give this function a more descriptive name. Something like removeDuplicateRooms.

File: apps/meteor/client/views/admin/rooms/RoomsTable.tsx [hunk 5]
```diff
@@ -205,9 +223,21 @@ const RoomsTable = ({ reload }: { reload: MutableRefObject<() => void> }): React
...
+		const set2 = new Set(array2);
+
+		return [...new Set(array1)].filter((x) => set2.has(x));
```
Comment (rique223) [inline apps/meteor/client/views/admin/rooms/RoomsTable.tsx:229]: Don't use single letter variable names.

File: packages/ui-client/src/components/MultiSelectCustom/MultiSelectCustom.tsx [hunk 6]
```diff
@@ -0,0 +1,138 @@
...
+	isGroupTitle: never;
+	checked: boolean;
+};
+
+// TODO: perhaps create a method to accept SelectOptions and convert to CheckboxOptionProp?
+// export type SelectOption = readonly [
+//     value: string, -> id
+//     label: string, -> text
+//     selected?: boolean -> checked
+// ];
```
Comment (rique223) [inline packages/ui-client/src/components/MultiSelectCustom/MultiSelectCustom.tsx:56]: If this type is not implementable right now, remove it please.

File: packages/ui-client/src/components/MultiSelectCustom/MultiSelectCustom.tsx [hunk 7]
```diff
@@ -0,0 +1,138 @@
...
+ * @param defaultTitle dropdown text before selecting any options (or all of them). For example: 'All rooms'
+	@param selectedOptionsTitle dropdown text after clicking one or more options. For example: 'Rooms (3)'
+ * @param selectedOptions array with clicked options. This is used in the useFilteredRooms hook, to filter the Rooms' table. This array joins all of the individual clicked options from all available MultiSelectCustom components in the page. It helps to create a union filter for all the selections.
+ * @param setSelectedOptions part of an useState hook to set the previous selectedOptions
+ * @param customSetSelected part of an useState hook to set the individual selected checkboxes from this instance.
+ * @returns a React Component that should be used with a custom hook for filters, such as useFilteredRooms.tsx.
+ * Check out the following files, for examples:
+ * 	useFilteredRooms.tsx,
+ * 	RoomsTable.tsx
+ */
```
Comment (rique223) [inline packages/ui-client/src/components/MultiSelectCustom/MultiSelectCustom.tsx:35]: Place this documentation above DropDownProps please.

**Review by rique223 — CHANGES_REQUESTED** [review]

**Review by rique223 — APPROVED** [review]
LGTM!

**Review by rique223 — APPROVED** [review]

**Pull request merged**